questions/src/main/java/com/ds/questions/aniket/BubbleSort.java

```diff
@@ -10,17 +9,34 @@
 //Example
 //Input : [5, 4, 11, 18, 1, 9]   Output: [1, 4, 5, 9, 11, 18]
 
+// ans: Time Complexity is O(n2)
+
 public class BubbleSort {
-    private static int[] inputData = {5, 4, 11, 18, 1, 9};
+    private static int[] inputData = { 5, 4, 11, 18, 1, 9 };
 
     public static void main(String[] args) {
-        getSortedArray(inputData);
+	int[] result = getSortedArray(inputData);
+	for (int i = 0; i < result.length; i++) {
+	    System.out.print(result[i] + " ");
+	}
     }
 
     private static int[] getSortedArray(int[] inputData) {
-
-
-        return null;
+	if (inputData == null || inputData.length == 0) {
+	    return new int[0];
+	}
+	if (inputData.length == 1) {
+	    return inputData;
+	}
+	for (int i = 0; i < inputData.length; i++) {
```

Comment:
i < inputData.length -1; is the right approach (I think), because the outer loop will run one extra loop for last index which is not required and exit after checking the inner loop.

Reply:
No, it wouldn't execute for last element. Inner loop initialization and condition is such that it wouldn't execute. j = i+1 and j < inputData.length

Reply:
Ok